api/vm/compile.ts

```diff
@@ -69,9 +69,20 @@ export function compile(compileConfig: dataform.ICompileConfig) {
 export function listenForCompileRequest() {
   process.on("message", (compileConfig: dataform.ICompileConfig) => {
     try {
-      const compiledResult = compile(compileConfig);
+      let compiledResult: string = compile(compileConfig);
       const writer = new net.Socket({ fd: 4 });
-      writer.write(compiledResult);
+      // max socket write length seems to be 2^12 (4096) bytes so split the writing.
+      let splitCompiledResult: string[] = [];
+      while (compiledResult.length > 0) {
+        splitCompiledResult.push(compiledResult.substring(0, 2 ** 12));
+        compiledResult = compiledResult.substring(2 ** 12);
+      }
+      for (let chunk of splitCompiledResult) {
+        writer.write(chunk, "utf8");
+        // This is needed for an unclear reason.
+        // tslint:disable-next-line: no-console
+        console.log();
```

Comment:
Any ideas why?

Reply:
are you sure the actual problem isn't that you're just not waiting for the write to complete? `socket.write` takes a callback which is called when the write completes, seems like you should `await` / `promisify`: https://nodejs.org/api/net.html#net_socket_write_data_encoding_callback